src/main/redux/sagas/app.ts

```diff
@@ -87,6 +87,18 @@ export function exit() {
         - quit
         */
 
+        const exitNow = () => {
+
+            // clean the db just before to quit
+            // tslint:disable-next-line: no-floating-promises
+            compactDb()
+                .finally(() => {
```

Comment:
I would like to suggest capturing all possibilities (note that the SQLite3 DB is *not* compacted even after the successful `compactDb()` `Promise.all()`):
```
            compactDb().then(() => {
                debug("Database(s) compacted successfully.");
            }).catch((err) => {
                debug("Database(s) compact failed:");
                debug(err);
            })
            .finally(() => {
                debug("EXIT NOW");
                app.exit(0);
            });
```

Reply:
I am now trying with `npm run start` to test LevelDown (SQLite3 is `npm run start:dev`).

Reply:
Tip: to test the `config` DB uncontrolled growth, simply play an audiobook for a few minutes :)

Reply:
Interesting: with LevelDown the config's `*.log` file grows (several MBs), and also does not get clean after `compact()`.

Reply:
The compaction code is invoked successfully:
https://github.com/rh389/pouchdb-adapter-websql-core/blob/e58ceaf4b73b7afcaffe695a76d88d1109c834af/src/index.js#L977-L1003

No `websqlError(callback)`, and the final `callback();` is invoked.

Reply:
Ah, the actual compaction routine is never called! `digestsToCheck` is empty:
https://github.com/rh389/pouchdb-adapter-websql-core/blob/e58ceaf4b73b7afcaffe695a76d88d1109c834af/src/utils.js#L92-L98

Reply:
The only `DELETE` is see when debugging is:
https://github.com/rh389/pouchdb-adapter-websql-core/blob/e58ceaf4b73b7afcaffe695a76d88d1109c834af/src/utils.js#L142-L144
...but the actual `config` SQLite3 DB file remains big.

Reply:
Okay so after analysing the SQLite3 database with a browser app, I can see that the `by-sequence` and `document-store` tables have been cleaned-up successfully, in the sense that there are no backup copies of the JSON data, but instead there are many `status` = `missing` references (and only a single `available` instance). I imagine that what takes space inside the binary file is the indexes? Not sure. At any rate, I am happy to merge this PR (once the `.then()` and `.catch()` are added)

Reply:
👍